### PR TITLE
Team select component

### PR DESCRIPTION
**Pull request by Setup1337**
https://imgur.com/a/hS3C9Ps
Source code contains comments for future development, if design will change

### CODE REVIEW

File: frontend/src/shared/components/tasque-team-select/tasque-team-select.component.html [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    <a class="user-avatar" *ngFor="let item of avatars.slice(0,7); let i = index" [attr.data-index]="i"
+       style="left: {{i*-18}}px;">
+      <img *ngIf="item.avatarURL" class="user-avatar-img" [src]="item.avatarURL" [alt]="item.username">
+      <div *ngIf="!item.avatarURL" class="user-no-avatar" style="background-color: {{profileColors[i]}}">
+        {{item.username[0]}}
+      </div>
+      <div class="user-overflow" *ngIf="i==6" (click)="showPopUp = !showPopUp">+{{avatars.length-7}}</div>
+    </a>
+  </div>
+<!--  <div class="add-users">-->
```
Comment (ismv) [inline frontend/src/shared/components/tasque-team-select/tasque-team-select.component.html]: don't leave unused code pls

File: frontend/src/shared/components/tasque-team-select/tasque-team-select.component.sass [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  box-sizing: border-box
+  font-family: 'Nunito', sans-serif
+  font-style: normal
+
+.container
+  display: flex
+  width: 265px
+  height: 45px
+
+
```
Comment (ismv) [inline frontend/src/shared/components/tasque-team-select/tasque-team-select.component.sass]: empty space

Comment (Setup1337) [reply frontend/src/shared/components/tasque-team-select/tasque-team-select.component.sass]: Fixed

File: frontend/src/shared/components/tasque-team-select/tasque-team-select.component.html [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+<div [hidden]="!showPopUp" class="pop-up">
+  <div class="user-list-container">
+      <div class="container-title">
+        <span>Members</span>
+        <a class="pop-up-close" (click)="showPopUp = !showPopUp">
+          <svg width="14" height="14" viewBox="0 0 24 23" fill="none" xmlns="http://www.w3.org/2000/svg">
+            <g opacity="0.4">
+              <path fill-rule="evenodd" clip-rule="evenodd" d="M18 6.76809L16.59 5.47559L11 10.5998L5.41 5.47559L4 6.76809L9.59 11.8923L4 17.0164L5.41 18.3089L11 13.1848L16.59 18.3089L18 17.0164L12.41 11.8923L18 6.76809Z" fill="#83858A" stroke="#83858A"/>
+            </g>
+          </svg>
```
Comment (Anatolii-Ivanov) [inline frontend/src/shared/components/tasque-team-select/tasque-team-select.component.html]: Move svg to a file

File: frontend/src/shared/components/tasque-team-select/tasque-team-select.component.sass [hunk 4]
```diff
@@ -0,0 +1,136 @@
+@import 'src/colors.sass'
+
+*
+  box-sizing: border-box
+  font-family: 'Nunito', sans-serif
+  font-style: normal
```
Comment (Anatolii-Ivanov) [inline frontend/src/shared/components/tasque-team-select/tasque-team-select.component.sass]: No global styles in the component file... Move it to global.css

**Review by Anatolii-Ivanov — CHANGES_REQUESTED** [review]

File: frontend/src/shared/components/tasque-team-select/tasque-team-select.component.html (hunk 3 above)
Comment (Setup1337) [reply frontend/src/shared/components/tasque-team-select/tasque-team-select.component.html]: Done

File: frontend/src/shared/components/tasque-team-select/tasque-team-select.component.sass (hunk 4 above)
Comment (Setup1337) [reply frontend/src/shared/components/tasque-team-select/tasque-team-select.component.sass]: Fixed

File: frontend/src/shared/components/tasque-team-select/models.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+﻿export interface UserCircle {
+  email: string,
+  username: string,
+  profileURL: string,
+  avatarURL: string | null,
```
Comment (Its-OP) [inline frontend/src/shared/components/tasque-team-select/models.ts]: avatarURL should not be nullable (here and everywhere else in the code). We will assign something by default (e.g. the first letter of the name)

File: frontend/src/shared/components/tasque-team-select/tasque-team-select.component.sass [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+  margin-left: 15px
+
+.add-users button
+  aspect-ratio: 1 / 1
+  height: 100%
+  border-radius: 100%
+  border: none
+  font-weight: 700
+  font-size: 18px
+  color: colorscheme(violet-200)
```
Comment (Its-OP) [inline frontend/src/shared/components/tasque-team-select/tasque-team-select.component.sass]: Why not to use tasque-button ?

File: frontend/src/shared/components/tasque-team-select/tasque-team-select.component.ts [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+  templateUrl: './tasque-team-select.component.html',
+  styleUrls: ['./tasque-team-select.component.sass']
+})
+export class TasqueTeamSelectComponent implements OnInit {
+
+  public inputSearch = '';
+
+  public showPopUp = false;
+
+  public avatars: UserCircle[] = [
```
Comment (Its-OP) [inline frontend/src/shared/components/tasque-team-select/tasque-team-select.component.ts]: Could you add marks on the code fragments, that should be deleted when the component gets integrated somewhere ?

**Review by Its-OP — CHANGES_REQUESTED** [review]

File: frontend/src/shared/components/tasque-team-select/tasque-team-select.component.html (hunk 3 above)
Comment (Its-OP) [reply frontend/src/shared/components/tasque-team-select/tasque-team-select.component.html]: Why not to use mat-icon or FA icon ?

File: frontend/src/shared/components/tasque-team-select/models.ts (hunk 5 above)
Comment (Setup1337) [reply frontend/src/shared/components/tasque-team-select/models.ts]: Made avatarURL not nullable

File: frontend/src/shared/components/tasque-team-select/tasque-team-select.component.ts (hunk 7 above)
Comment (Setup1337) [reply frontend/src/shared/components/tasque-team-select/tasque-team-select.component.ts]: Added marks for avatars array, that should be deleted

File: frontend/src/shared/components/tasque-team-select/tasque-team-select.component.sass (hunk 6 above)
Comment (Setup1337) [reply frontend/src/shared/components/tasque-team-select/tasque-team-select.component.sass]: Removed button at all

File: frontend/src/shared/components/tasque-team-select/tasque-team-select.component.html [hunk 8]
```diff
@@ -0,0 +1,51 @@
+<div class="container">
+  <div class="users" *ngIf="avatars.length <= 4">
+    <a class="user-avatar" *ngFor="let item of avatars.slice(0,7); let i = index" [attr.data-index]="i">
+      <img *ngIf="item.avatarURL" class="user-avatar-img" [src]="item.avatarURL" [alt]="item.username">
+      <div *ngIf="!item.avatarURL" class="user-no-avatar" style="background-color: {{profileColors[i]}}">
```
Comment (Its-OP) [inline frontend/src/shared/components/tasque-team-select/tasque-team-select.component.html]: This ngIf condition would always be false, as we always have value in avatarURL. There were a couple more such conditions on this page.
Styles for .user-no-avatar should be deleted as well

**Review by Its-OP — CHANGES_REQUESTED** [review]

File: frontend/src/shared/components/tasque-team-select/tasque-team-select.component.html (hunk 8 above)
Comment (Setup1337) [reply frontend/src/shared/components/tasque-team-select/tasque-team-select.component.html]: Actually no, this works fine 
image

**Review by Anatolii-Ivanov — APPROVED** [review]

**Review by FeloniukM — APPROVED** [review]

**Review by Its-OP — APPROVED** [review]

**Pull request merged**